gitlab/resource_gitlab_project.go

```diff
@@ -208,6 +208,22 @@ var resourceGitLabProjectSchema = map[string]*schema.Schema{
 		Type:     schema.TypeBool,
 		Optional: true,
 	},
+	"template_name": {
+		Type:     schema.TypeString,
+		Optional: true,
+	},
+	"template_project_id": {
+		Type:     schema.TypeInt,
+		Optional: true,
+	},
```

Comment:
You can enforce these to be mutually exclusive using `ConflictsWith` ([source](https://github.com/hashicorp/terraform-plugin-sdk/blob/35d7fcb6c861606651dee1584261138dd71fb61b/helper/schema/schema.go#L196)).

```suggestion
	"template_name": {
		Type:          schema.TypeString,
		Optional:      true,
		ConflictsWith: []string{"template_project_id"},
	},
	"template_project_id": {
		Type:          schema.TypeInt,
		Optional:      true,
		ConflictsWith: []string{"template_name"},
	},
```

Then you can remove your `if options.TemplateName != nil && options.TemplateProjectID != nil` check in the create function.